src/include/gcc.h

```diff
@@ -280,6 +280,23 @@ WT_ATOMIC_FUNC(size, size_t, size_t *vp, size_t v)
         __asm__ volatile("" ::: "memory"); \
     } while (0)
 
+
+#elif defined(__riscv) && (__riscv_xlen == 64)
+#define WT_PAUSE() __asm__ volatile("nop" ::: "memory")
+#define WT_FULL_BARRIER()                               \
```

Comment:
This file lacks a detailed explanation of what the intended semantics of `WT_FULL_BARRIER`, `WT_READ_BARRIER`, and `WT_WRITE_BARRIER` really are.

However, since the x86_64 mapping is:
- `FULL` : `mfence`
- `READ`: `lfence`
- `WRITE`: `sfence`

My understanding is that `lfence` and `sfence` only synchronize with other `[lm]fence` and `[sm]fence` instructions. Reading https://five-embeddev.com/riscv-isa-manual/latest/memory.html#sec:mm:fence, it seems then that on riscv where we can specify the predecessor set and successor set for the fence, that we should have:
- `FULL`: `rw, rw`
- `READ`: `r, r`
- `WRITE`: `w, w`

I'd like to dig a little deeper to make sure that these are the right `fence` intsructions in general, but it seems plausible.

Reply:
Agree, this makes sense to me.

Reply:
Actually, I found that the file does have a good explanation, I just skimmed past it. Based on the description, I'm still more confident that `r, r` correctly models `READ` and `w, w` is correct for `WRITE`. I've added a comment with a link, it will show up on the next push.